python/pyspark/ml/param/__init__.py

```diff
@@ -65,6 +75,146 @@ def __eq__(self, other):
             return False
 
 
+class TypeConverters(object):
+    """
+    .. note:: DeveloperApi
+
+    Factory methods for common type conversion functions for `Param.typeConverter`.
+
+    .. versionadded:: 2.0.0
+    """
+
+    @staticmethod
+    def _is_numeric(value):
+        vtype = type(value)
+        return vtype in [int, float, np.float64, np.int64] or vtype.__name__ == 'long'
+
+    @staticmethod
+    def _is_integer(value):
+        if TypeConverters._is_numeric(value):
+            value = float(value)
+            return value.is_integer()
+        else:
+            return False
+
+    @staticmethod
+    def _can_convert_to_list(value):
+        vtype = type(value)
+        return vtype == list or vtype == np.ndarray or isinstance(value, Vector)
+
+    @staticmethod
+    def _is_string(value):
```

Comment:
I'd follow existing conventions.  Look at, e.g., the beginning of file feature.py:
```
import sys
if sys.version > '3':
    basestring = str
```
Then you can use ```isinstance(blah, basestring)```